agent/src/main/java/com/testsigma/agent/mobile/ios/IosDeviceListener.java

```diff
@@ -133,13 +138,23 @@ public void removeDeviceListenerCallback() {
       } catch (Exception e) {
         usbMuxSocket = null;
       }
+      isStarted = false;
     }
-    isStarted = false;
   }
 
-  public MobileDevice getMobileDevice(String uniqueId) throws TestsigmaException {
-    MobileDevice mobileDevice = new MobileDevice();
+  public void getInitialSimulatorDevices() {
+    try {
+      List<MobileDevice> devices = iosDeviceService.simulatorDeviceList();
+      for (MobileDevice device : devices) {
+        this.addDevice(device);
+      }
+    } catch(Exception e) {
+      log.error(e.getMessage());
```

Comment:
Change this to log.error("Error in fetching Simulator devices",e); Its better to have more info

Reply:
Will do that